cyprecice/cyprecice.pyx

```diff
@@ -1211,6 +1211,223 @@ cdef class Interface:
         self.thisptr.readScalarData (data_id, vertex_id, _value)
         return _value
 
+    def write_block_vector_gradient_data (self, data_id, vertex_ids, gradientValues):
+        """
+        Writes vector gradient data given as block. This function writes values of specified vertices to a dataID.
+        Values are provided as a block of continuous memory. Values are stored in a numpy array [N x D] where N = number
+        of vertices and D = dimensions of geometry squared.
+
+        Parameters
+        ----------
+        data_id : int
+            Data ID to write to.
+        vertex_ids : array_like
+            Indices of the vertices.
+        gradientValues : array_like
+             Gradient values differentiated in the spacial direction (dx, dy) for 2D space, (dx, dy, dz) for 3D space
```

Comment:
From what you write above, the `gradientValues` array should have 'number of gradient components' many entries. Thus, there is missing information about the format, right?

Reply:
Format is shown in examples, writing both scalar and vector format seems to just replicate the examples below https://github.com/kursatyurt/python-bindings/blob/bd7d0315c7db73b996a7879944c05ef645ca5a5c/cyprecice/cyprecice.pyx#L1237-L1250